### PR TITLE
New Twitter Wall

### PR DESCRIPTION
**Pull request by aayusharora**
The pull request includes three directives:
1. background-directive.js :It is for the background of Twitter Wall which changes whenever anybody tweets on @FOSSASIA .
2.SingleTweet.js:This directive generates the slideshow of tweets .
3.Width.js replaces the broken images from Twitter with a hardcoded image.
4.Twitterwall.css is the styling file
5. Some changes are there in controller (TwitterController.js) and index.html.

### CODE REVIEW

**Comment by aayusharora** [comment]
@creativepsyco ,@mariobehling,@mananwason,Kindly review the new Twitter wall.

File: app/components/twitterwall/TwitterwallController.js [hunk 1]
```diff
@@ -12,13 +8,23 @@ twitModule.config(['$stateProvider', function($stateProvider) {
...
-    ]
-);
\ No newline at end of file
+/*created By aayusharora on 17/2/16*/
+twitModule.controller('TwitterwallController',
+    ['$state','$http','$scope',function($state,$http,$scope){
+      $scope.fetched=false;
+        $http({
+  method: 'GET',
+  url: 'http://loklak.org/api/search.json?q=fossasia',
```
Comment (championswimmer) [inline app/components/twitterwall/TwitterwallController.js]: Should take the keyword from a config.json or something like that. Should not be hardcoded.

**Comment by aayusharora** [comment]
I will do that @championswimmer ,thanks

**Comment by aayusharora** [comment]
Here is the working model:
twitterwall

**Comment by aayusharora** [comment]
@championswimmer ,review ,its done 
New Service added : TwitterJsonFactory.js
changed file:1.config.js: Url added
                    2 .Twittercontroller.js : for removing hard coded url

**Comment by aayusharora** [comment]
@championswimmer ,@mariobehling,Kindly help me by reviewing this .

**Comment by championswimmer** [comment]
It looks good for now. Let me test it locally, and if all is good, will merge it

**Comment by championswimmer** [comment]
could you rebase it though? it got mashed up with @creativepsyco 's commit.

**Comment by aayusharora** [comment]
@championswimmer ,Its done ,Check please.

**Comment by championswimmer** [comment]
You're not the author of the commit it seems and it's mashed up with @daminisatya 's commit now

**Comment by aayusharora** [comment]
@championswimmer ,its done now ,I suppose. :)

File: app/components/common/ApiJsonFactory.js [hunk 2]
```diff
@@ -23,8 +22,9 @@ var baseUrl = (config.use_testApi?'testapi/':config.apiBaseGetUrl) +
                 }, function (error) {
                     console.error(error);
                 });
-
+
```
Comment (championswimmer) [inline app/components/common/ApiJsonFactory.js]: do not make just whitespace changes.

File: app/components/twitterwall/background-directive.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            for(var photo in scope.array){
+                if(scope.array[photo].user.profile_image_url_https){
+                    var obj={
+                        photo:scope.array[photo].user.profile_image_url_https,
+                        id:scope.array[photo].user.user_id
+                    }
+                    scope.photos.push(obj);
+                  }
+            }
+        })
```
Comment (championswimmer) [inline app/components/twitterwall/background-directive.js]: @aayusharora fix that

File: app/components/twitterwall/background-directive.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        changeBackground();
+
+        function getBackground(){
+            scope.$watch('array',function(oldvalue,newvalue){
+            for(var photo in scope.array){
+                if(scope.array[photo].user.profile_image_url_https){
+                    var obj={
+                        photo:scope.array[photo].user.profile_image_url_https,
+                        id:scope.array[photo].user.user_id
+                    }
```
Comment (championswimmer) [inline app/components/twitterwall/background-directive.js]: @aayusharora fix that

File: index.html [hunk 5]
```diff
@@ -4,11 +4,13 @@
...
     <!-- web app manifest: https://w3c.github.io/manifest/ -->
     <link rel="manifest" href="manifest.webmanifest">
-    
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
+    <link href='https://fonts.googleapis.com/css?family=Roboto' rel='stylesheet' type='text/css'>
     <script src="config.js"></script>
 
     <!-- inject:css -->
     <link rel="stylesheet" href="assets/css/leaflet.css">
+    <link rel="stylesheet" href="assets/css/twitterwall.css">
```
Comment (championswimmer) [inline index.html]: add these to gulpfile too

File: index.html [hunk 6]
```diff
@@ -58,13 +60,17 @@ <h1 id="title"> {{ appCtrl.Event.name }} </h1>
...
+    <script src="app/components/common/TwitterJsonFactory.js"></script>
     <script src="app/components/map/MapController.js"></script>
     <script src="app/components/sessions/SessionsController.js"></script>
     <script src="app/components/sidenav/SidenavController.js"></script>
     <script src="app/components/speakers/SpeakersController.js"></script>
     <script src="app/components/sponsors/SponsorsController.js"></script>
-    <script src="app/components/tracks/TracksController.js"></script>
     <script src="app/components/twitterwall/TwitterwallController.js"></script>
+    <script src="app/components/twitterwall/background-directive.js"></script>
+    <script src="app/components/twitterwall/singletweet.js"></script>
```
Comment (championswimmer) [inline index.html]: these to gulpfile too

Comment (aayusharora) [inline index.html]: @championswimmer ,

Help me in this ,seems they are already there

**Comment by aayusharora** [comment]
@championswimmer , I have performed gulp cleanup ,my files are injecting automatically as they are in app/components and assets folder for which gulp task are already written .
Seems fine now.

**Comment by aayusharora** [comment]
Hi,@championswimmer, I am planning to add some more functionality into this.
1. Making the text renderable.
2. Dynamic background ,already background update with new tweets ,want to work little on its UX.
It will be better if you check this before I move ahead .

**Comment by championswimmer** [comment]
You need to rebase to master first.

**Comment by championswimmer** [comment]
This is looking good other than that. Would love to have this twitterwall. Looks nice

**Comment by aayusharora** [comment]
@championswimmer ,check and merge please.

**Comment by aayusharora** [comment]
@championswimmer ,@mariobehling ,kindly review ,so that I can move ahead.

File: app/components/twitterwall/TwitterwallController.js [hunk 7]
```diff
@@ -1,7 +1,3 @@
-/**
- * Created by championswimmer on 9/8/15.
```
Comment (championswimmer) [inline app/components/twitterwall/TwitterwallController.js]: Just add your attribution below it :D

**Pull request merged**